### PR TITLE
JS-28:  Implement routing and routing security

### PR DESCRIPTION
**Pull request by TeodorTunev**

### CODE REVIEW

File: src/app/services/auth-manager.service.ts [hunk 1]
```diff
@@ -1,17 +1,60 @@
-import {Injectable} from '@angular/core';
-import {LocalStorageWrapper} from './local-storage-wrapper';
+import {Injectable} from "@angular/core";
+import {ApiCall} from "./api-call";
+import "rxjs/add/operator/map";
```
Comment (belev) [inline src/app/services/auth-manager.service.ts]: Extract to 'rxjs-extensions.ts' with the used imports.
Import it to app.module.ts

File: src/app/services/auth-manager.service.ts [hunk 2]
```diff
@@ -1,17 +1,60 @@
...
 
   isUserLoggedin(): boolean {
     return !!this.user;
   }
+
+  logUser(email: string, password: string) {
+    return this.apiCall.post('users/sessions', {
+      "data": {
+        "attributes": {
+          "email": email,
```
Comment (belev) [inline src/app/services/auth-manager.service.ts]: email-or-phone

File: src/app/services/auth-manager.service.ts [hunk 3]
```diff
@@ -1,17 +1,60 @@
...
+        this.localStorageWrapper.setObject(this.storageAuthorizationData, result);
+        return this.userProxy.getUser(result['user-id']);
+      })
+      .then(response => {
+        return this.handleUserResult(response);
+      });
+  }
+
+  handleUserResult(response) {
+    console.log('response');
```
Comment (belev) [inline src/app/services/auth-manager.service.ts]: remove console.log

**Comment by buren** [comment]
👍

**Pull request merged**